src/internal/connector/graph_connector_onedrive_test.go

```diff
@@ -139,13 +185,150 @@ var (
 	}
 )
 
-func withItems(items ...[]itemInfo) []itemInfo {
-	res := []itemInfo{}
-	for _, i := range items {
-		res = append(res, i...)
+func newOneDriveCollection(
+	t *testing.T,
+	pathElements []string,
+	backupVersion int,
+) *onedriveCollection {
+	return &onedriveCollection{
+		pathElements:  pathElements,
+		backupVersion: backupVersion,
+		t:             t,
 	}
+}
+
+type onedriveCollection struct {
+	pathElements  []string
+	items         []itemInfo
+	aux           []itemInfo
+	backupVersion int
+	t             *testing.T
+}
 
-	return res
+func (c onedriveCollection) collection() colInfo {
+	return colInfo{
+		pathElements: c.pathElements,
+		category:     path.FilesCategory,
+		items:        c.items,
+		auxItems:     c.aux,
+	}
+}
+
+func (c *onedriveCollection) withFile(
+	name string,
+	fileData []byte,
+	user string,
+	roles []string,
+) *onedriveCollection {
+	switch c.backupVersion {
+	case 0:
+		// Lookups will occur using the most recent version of things so we need
+		// the embedded file name to match that.
+		c.items = append(c.items, onedriveItemWithData(
+			c.t,
+			name,
+			name+onedrive.DataFileSuffix,
+			fileData))
+
+	case 1:
+		fallthrough
+	case 2:
+		c.items = append(c.items, onedriveItemWithData(
+			c.t,
+			name+onedrive.DataFileSuffix,
+			name+onedrive.DataFileSuffix,
+			fileData))
+
+		metadata := onedriveMetadata(
+			c.t,
+			"",
+			name+onedrive.MetaFileSuffix,
+			user,
+			roles)
+		c.items = append(c.items, metadata)
+		c.aux = append(c.aux, metadata)
+
+	default:
```

Comment:
I was going to have this for when we start saving file names in metadata files. I should remove it from this PR though since it's not actually relevant. I probably added it by mistake 😅